### DISCUSSION TITLE
Resolving of hierarchical project structures

### DISCUSSION THREAD

**Discussion by Carsten-Berndt** [DependencyTrack/dependency-track#1253] [Ideas]
We have projects which are 1st party components in other projects. It would be very useful if Dependency Track could recognize that a project uses 1st party components that are described in their own project and use that information to provide insight on inherited dependencies.
There also should be a way to filter out inherited dependencies.

I believe this would allow for a much cleaner management of complex projects as opposed to creating large merged SBOMs with all 1st party dependencies resolved before uploading.

**Comment by stevespringett** [comment]
Related to #84 and #869. Both tickets are marked as 'help wanted'.

**Reply by Carsten-Berndt** [reply]
Oh, sorry. I looked in the discussions but forgot to check the issues.
Thanks for the reply ^^
